jctools-experimental/src/main/java/org/jctools/stacks/WFIStack.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jctools.stacks;
+
+import org.jctools.util.UnsafeAccess;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * A Wait-Free Intrusive Stack (Last-In, First-Out).
```

Comment:
Do you think would be nice to implement an additional `default` non-intrusive version (a-la CLQ) based on this for lazy users?

Reply:
What's the CLQ you're referring to?

Reply:
Sorry, you're right, I've missed to translate the acronym: `ConcurrentLinkedQueue` of the j.u.c